src/SideNavigation/SideNavigation.js

```diff
@@ -1,204 +1,251 @@
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
-import { BrowserRouter, Link } from 'react-router-dom';
 import React, { Component } from 'react';
 
-export const SideNav = props => {
-    const { icons, children, ...rest } = props;
-
-    const sideNavClasses = classnames(
-        'fd-side-nav',
-        {
-            'fd-side-nav--icons': icons
-        }
-    );
-
-    return (
-        <nav {...rest} className={sideNavClasses}>
-            {children}
-        </nav>
-    );
-};
-SideNav.propTypes = {
-    children: PropTypes.node,
-    icons: PropTypes.bool
-};
-
-SideNav.propDescriptions = {
-    icons: 'Set to **true** enables side navigation collapsed with icons.'
-};
-
-export class SideNavList extends Component {
+export class SideNav extends Component {
     constructor(props) {
         super(props);
 
-        let initialState = [];
-
-        props.items.forEach(item => {
-            if (item.hasChild) {
-                let id = item.id;
-                let obj = {};
-
-                obj[id] = false;
-                initialState.push(obj);
-            }
-        });
-
         this.state = {
-            selectedItem: 'item_2',
-            itemStates: initialState
+            selectedId: props.selectedId
```

Comment:
I think we will need a `getDerivedStateFromProps` (previously known as `componentWillReceiveProps`) to keep the internal state in sync since `selectedId ` is a prop.

Reply:
Corrected in ca228a2